src/Beacon.jsx

```diff
@@ -11,7 +11,20 @@ const TOOLTIP_OVERLAY_CLASS = 'tour-overlay';
 const TOOLTIP_FADED_CLASS = 'tour-faded';
 const TARGET_CLONE_ID = 'tour-target-clone';
 // If margins are within tolerance then we center tooltip
-const TOOLTIP_TOLERANCE = 50;
+// Tolerance is ratio of margins to total dimension (height or width)
+// So if the difference between the tooltip margins is less than
+// `dimension/4` then we center it.
+const TOOLTIP_TOLERANCE_RATIO = 4;
```

Comment:
I used to use a fixed tolerance (50px) but it was too small and should adjust for the size of the dimension since we want more tolerance in the vertical direction if the display is in portrait orientation.